src/core/net/srp_server.cpp

```diff
@@ -1870,7 +1840,18 @@ Error Server::Host::MergeServicesAndResourcesFrom(Host &aHost)
 
 const Server::Service::Description *Server::Host::FindServiceDescription(const char *aInstanceName) const
```

Comment:
```suggestion
const RetainPtr<Server::Service::Description> Server::Host::FindServiceDescription(const char *aInstanceName) const
```
I think it's better to avoid using raw pointer for resources which are managed by smart pointers,
Otherwise, the caller may refer to a resource which can be auto released. Thoughts?

Reply:
Thanks. Good suggestion. 👍 
 I think it is a good way to use this model (though it may be fine in the case here for now, but I agree it is cleaner and safer model to adopt the model to return `RetainPtr` back to caller if we use in internal data structure).

Will change in next push.

Reply:
Updated this in new push. 

Also added a second smaller commit on this PR which adds `Ptr`(a simpler wrapper over a pointer) and used as the base class of both `RetainPtr` and `OwnedPtr`. This provides the common simple methods (avoiding repeating
the same code in each smart pointter).

Reply:
Thanks @abtink , the second commit also looks good to me!